### ISSUE TITLE
AS3 container async implementation

### ISSUE THREAD

**Issue by kreynoldsf5** [F5Networks/f5-appsvcs-extension#79]
##### Do you already have an issue opened with F5 support?
No.

##### ISSUE TYPE

 - Bug Report

##### AS3 BUILD/ VERSION

```
f5devcentral/f5-as3-container, tag > 3.7.0
```

##### BIGIP VERSION

```
NA
```

##### SUMMARY

```
"async=true" workflow appears to be improperly implemented in f5devcentral/f5-as3-container. Submitting a declaration using "async=true" returns JSON like:

{
  "results": "Declaration successfully submitted",
  "id": "26d44633-6f65-47b9-ad75-3a4380132606",
  "selfLink": "https://localhost/mgmt/shared/appsvcs/task/26d44633-6f65-47b9-ad75-3a4380132606"
}

The task endpoint does not exist in the container or the selflink doesn't properly describe the endpont. In AS3 >3.7.0 where async occurs after 45 seconds automatically this will be especially problematic.
```

##### STEPS TO REPRODUCE

```bash
#!/bin/sh
post_result=$(curl -sk -H "Content-Type: application/json" -X POST -d @dry_declaration.json 'https://f5devcentral-f5-as3-container:443/mgmt/shared/appsvcs/declare?async=true')
task_id=$(echo "${post_result}" | jq -r '.id')
task_result=$(curl -sk https://f5devcentral-f5-as3-container:443/mgmt/shared/appsvcs/task/${task_id})
all_tasks_result=$(curl -sk https://f5devcentral-f5-as3-container:443/mgmt/shared/appsvcs/task)
echo $post_result
echo $task_result
echo $all_tasks_result
```
```json
{"results":"Declaration successfully submitted","id":"df2085d6-e890-404a-931f-78b5b0ac1fa2","selfLink":"https://localhost/mgmt/shared/appsvcs/task/df2085d6-e890-404a-931f-78b5b0ac1fa2"}
{"message":"Request record does not exist! Record id: df2085d6-e890-404a-931f-78b5b0ac1fa2"}
{"message":"An error occurred while retrieving async records: undefined"}
```

##### EXPECTED RESULTS

```
Ideally, the async task endpoint would exist in the container.
```

##### ACTUAL RESULTS

```
Task endpoint not properly implemented in the container.
```

**Comment by dstokesf5** [comment]
Thank you for the bug report. We actually already have this fixed for the upcoming 3.10 release.

**Comment by kaustriaf5** [comment]
Tracked internally as 878.

@kreynoldsf5,
Please see release [3.10](https://github.com/F5Networks/f5-appsvcs-extension/releases/tag/v3.10.0) for the fix. Closing this issue.
